src/main/java/org/reaktivity/nukleus/mqtt/internal/stream/MqttServerFactory.java

```diff
@@ -265,12 +300,375 @@ private RouteFW resolveTarget(
+        final long budgetId,
+        final DirectBuffer buffer,
+        final int offset,
+        final int limit)
+    {
+        final MqttSubscribeFW subscribe = mqttSubscribeRO.tryWrap(buffer, offset, limit);
+        int progress = limit;
+        if (limit == 0 || subscribe == null)
+        {
+            server.doEncodeDisconnect(authorization, 0x82);
+            server.doNetworkEnd(traceId);
+        }
+        else
+        {
+            server.onDecodeSubscribe(authorization, subscribe);
+            progress = subscribe.limit();
+        }
+
+        server.decoder = decodePacketType;
+        return progress;
+    }
+
+    private int decodeUnsubscribe(
+        MqttServer server,
+        final long traceId,
+        final long authorization,
+        final long budgetId,
+        final DirectBuffer buffer,
+        final int offset,
+        final int limit)
+    {
+        final MqttUnsubscribeFW unsubscribe = mqttUnsubscribeRO.tryWrap(buffer, offset, limit);
+        int progress = limit;
+        if (limit == 0 || unsubscribe == null)
+        {
+            server.doEncodeDisconnect(authorization, 0x82);
+            server.doNetworkEnd(traceId);
+        }
+        else
+        {
+            server.onDecodeUnsubscribe(authorization, unsubscribe);
+            progress = unsubscribe.limit();
+        }
+
+        server.decoder = decodePacketType;
+        return progress;
+    }
+
+    private int decodePingreq(
+        MqttServer server,
+        final long traceId,
+        final long authorization,
+        final long budgetId,
+        final DirectBuffer buffer,
+        final int offset,
+        final int limit)
+    {
+        final MqttPingReqFW ping = mqttPingReqRO.tryWrap(buffer, offset, limit);
+        server.onDecodePingReq(authorization, ping);
+        server.decoder = decodePacketType;
+        return ping.limit();
+    }
+
+    private int decodeDisconnect(
+        MqttServer server,
+        final long traceId,
+        final long authorization,
+        final long budgetId,
+        final DirectBuffer buffer,
+        final int offset,
+        final int limit)
+    {
+        final MqttDisconnectFW disconnect = mqttDisconnectRO.tryWrap(buffer, offset, limit);
+        int progress = limit;
+        if (limit == 0 || disconnect == null)
+        {
+            server.doEncodeDisconnect(authorization, 0x82);
+            server.doNetworkEnd(traceId);
+        }
+        else
+        {
+            server.onDecodeDisconnect(traceId, disconnect);
+            progress = disconnect.limit();
+        }
+
+        server.decoder = decodePacketType;
+        return progress;
+    }
+
+    private int decodeIgnoreOne(
+        MqttServer server,
+        long traceId,
+        long authorization,
+        long budgetId,
+        DirectBuffer buffer,
+        int offset,
+        int limit)
+    {
+        final MqttPacketFixedHeaderFW packet = mqttPacketFixedHeaderRO.tryWrap(buffer, offset, limit);
```

Comment:
Use `wrap` instead of `tryWrap` because `decodePacketType` guarantees a successful wrap.